ts/input/tex/TexParser.ts

```diff
@@ -195,6 +203,12 @@ export default class TexParser {
    * @param {StackItem|MmlNode} arg The new item.
    */
   public Push(arg: StackItem | MmlNode) {
+    if (arg instanceof BaseItem) {
+      arg.startI = this.saveI.pop();
+      this.saveI.push(arg.startI);
+      arg.stopI = this.i;
+      arg.startStr = this.string;
+    }
```

Comment:
If you change the `saveI` as suggested above, then this would become

``` js
    if (arg instanceof BaseItem) {
      arg.startI = this.saveI;
      arg.stopI = this.i;
      arg.startStr = this.string;
    }
```

No need to pop and then push the value.

If you don't want to do that change, then at least use

``` js
    arg.startI = this.saveI[this.saveI.length - 1];
```

instead of popping and then re-pushing the same value.